lhotse/cut/base.py

```diff
@@ -506,6 +506,93 @@ def trim_to_supervisions(
             cuts.append(trimmed)
         return CutSet.from_cuts(cuts)
 
+    def trim_to_alignments(
+        self,
+        type: str,
+        max_pause: Seconds = 0.0,
+        delimiter: str = " ",
+        keep_all_channels: bool = False,
+    ) -> "CutSet":  # noqa: F821
+        """
+        Splits the current :class:`.Cut` into its constituent alignment items (:class:`.AlignmentItem`).
+        These cuts have identical start times and durations as the alignment item. Additionally,
+        the `max_pause` option can be used to merge alignment items that are separated by a pause
+        shorter than `max_pause`.
+
+        For the case of a multi-channel cut with multiple alignments, we can either trim
+        while respecting the supervision channels (in which case output cut has the same channels
+        as the supervision) or ignore the channels (in which case output cut has the same channels
+        as the input cut).
+
+        .. hint:: If the resulting trimmed cut contains a single supervision, we set the cut id to
+            the ``id`` of this supervision, for better compatibility with downstream tools, e.g.
+            comparing the hypothesis of ASR with the reference in icefall.
+
+        .. hint:: If a MultiCut is trimmed and the resulting trimmed cut contains a single channel,
+            we convert it to a MonoCut.
+
+        :param type: The type of the alignment to trim to (e.g. "word").
+        :param max_pause: The maximum pause allowed between the alignments to merge them.
+        :param delimiter: The delimiter to use when joining the alignment items.
+        :param keep_all_channels: If ``True``, the output cut will have the same channels as the input cut. By default,
+            the trimmed cut will have the same channels as the supervision.
+        :param num_jobs: Number of parallel workers to process the cuts.
+        :return: a CutSet object.
+        """
+        from lhotse.supervision import AlignmentItem
+
+        # For the implementation, we first create new supervisions for the cut, and then
+        # use the `trim_to_supervisions` method to do the actual trimming.
+        new_supervisions = []
+        for segment in self.supervisions:
+            if (
+                segment.alignment is None
+                or type not in segment.alignment
+                or not segment.alignment[type]
+            ):
+                continue
+            alignments = sorted(segment.alignment[type], key=lambda a: a.start)
+
+            # Merge the alignments if needed. We also keep track of the indices of the
+            # merged alignments in the original list. This is needed to create the
+            # `alignment` field in the new supervisions.
+            merged_alignments = [(alignments[0], [0])]
+            for i, item in enumerate(alignments[1:]):
+                prev_item, prev_indices = merged_alignments[-1]
+                if item.start - prev_item.end <= max_pause:
```

Comment:
A catch here is that if there are alignment items which are continuous (i.e. `prev_end == cur_start`) then they will necessarily get merged. So in order to trim such that we get #cuts = num. of alignment items, a negative value of `max_pause` will need to be passed.